### ISSUE TITLE
Integrate babel-blade into Gatsby

### ISSUE THREAD

**Issue by stephanschubert** [gatsbyjs/gatsby#10066]
## Summary

Solve the Double Declaration problem with inline GraphQL.

**[7-min talk](https://youtu.be/30wOsJOluA4?t=497)** explains more.

[![babel blade walkthrough on youtube](https://user-images.githubusercontent.com/6764957/48116756-ef34a800-e21b-11e8-9d4f-049362c25b23.gif)](https://www.youtube.com/watch?v=z9wKcRjNqlw)

## Examples

* [AST Explorer](https://astexplorer.net/#/gist/01983f61e310f1eaf6b12a221556a937/c5fad39a809e238657ebaa002c5cdba7b2e5bf1e)
* [AST Explorer + Fragments](https://astexplorer.net/#/gist/01983f61e310f1eaf6b12a221556a937/f53fd8fd214daa57b1180e2c8a9a43806e088a94)
* [Babel Repl](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=JYWwDg9gTgLgBAbzgYQgOzQUwMYwDRwCOArplAJ5wC-cAZlBCHAOTFSEA2zA3AFC_Z0AZ3ggIAN2CYAiqQpwAvHGxRMAQxgy55ABQBKPoLQi4AWQlTFcHQmAATKnsUA-a7zhwAPHeDjn7jy9UDBwYAMCibQUEEjJdMUktOL0qcMDsAAtgDjtVNGibODsNNWonBVcENIijEwARAEEAFQarBKlZOJ1imDU9avTheAQYYBgOTAI1XGghGiVGloA6dswdAAN7AC44ABJbB3X-iIjVGDY0NxOT719_a-vPDIAmZxGxiapPAHoX-4eIghpjBZis1GAdMDoC44FCoEs0GoQJgUgMIj8fH40XADAMqKkIt9_hi7rwDEA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=&prettier=true&targets=&version=6.26.0&envVersion=) (Add `babel-plugin-blade` as plugin)

## Links

* `babel-blade` [Github repository](https://github.com/sw-yx/babel-blade)
* [Docs](https://babel-blade.netlify.com/docs/)

**Comment by LekoArts** [comment]
Could you please elaborate more thoroughly on why Gatsby would need that?

**Comment by stephanschubert** [comment]
@LekoArts It would dissolve the need for writing GraphQL queries manually (and keeping them up to date/in sync with the actual usage).

**Comment by jgierer12** [comment]
While this is a very interesting project that I'm closely following myself, I think it's a) far too early to use in production and b) completely unnecessary to integrate into Gatsby:

First off, it currently doesn't even work with Gatsby's `graphql` tag - see sw-yx/babel-blade#20. Additionally, it's not yet compatible with the way Gatsby handles fragments: sw-yx/babel-blade#17. There are also a bunch of other issues that need to be tackled before it's ready to use in any production code.

Besides, Gatsby already includes the `babel-plugin-macro`. Meaning that, once the above issues are fixed, `babel-blade` will be usable in Gatsby just by installing `blade.macro`. This is an ideal way to easily opt-in to `babel-blade` without having to change anything on Gatsby's side.

Once the project is in a state where it's comfortably usable within Gatsby, it would be a good idea to publish a blog post or guide for using the two together. I think `babel-blade` has a huge potential, especially combined with Gatsby, but it's just not there yet in terms of stability and compatibility.

**Comment by stephanschubert** [comment]
@jgierer12 I agree it's too early but I wanted to put it on the radar (didn't find any issue/feature request so far about it).
